File: public_encaptulation.cpp
#include<bits/stdc++.h>
using namespace std;

class Bank {
private:
    string holder_name;
    string branch;
    int balance;

public:
    Bank(const string& holder_name, int initial_deposit) : holder_name(holder_name), branch("banani 11"), balance(initial_deposit) {}

    void deposit(int amount) {
        balance += amount;
    }

    int get_balance() const {
        return balance;
    }

    int withdraw(int amount) {
        if (amount < balance) {
            balance -= amount;
            return amount;
        } else {
            cout << "Forkia taka nai" << endl;
            return 0;
        }
    }

    int get_balance_public() const {
        return balance;
    }
};

int main() {
    Bank rafsun("Choooto bro", 10000);

    cout << rafsun.get_balance() << endl;
    rafsun.deposit(40000);
    cout << rafsun.get_balance() << endl;
    cout << rafsun.get_balance_public() << endl;

    return 0;
}

File: public_encaptulation.py
class Bank:
    def __init__(self, holder_name, initial_deposit) -> None:
        self.holder_name = holder_name 
        self._branch = 'banani 11'  
        self.__balance = initial_deposit  

    def deposit(self, amount):
        self.__balance += amount

    def get_balance(self):
        return self.__balance
    
    def withdraw(self, amount):
        if amount < self.__balance:
            self.__balance -= amount
            return amount
        else:
            return 'Forkia taka nai'

    
    def get_balance_public(self):
        return self.__balance

rafsun = Bank('Choooto bro', 10000)

print(rafsun.holder_name)
rafsun.holder_name = 'boro vai'
rafsun.deposit(40000)
print(rafsun.get_balance())
print(rafsun.holder_name)

print(rafsun.get_balance_public())
